Tests/Runtime/Reporter/ReportUploadGuardServiceTests.cs

```diff
@@ -0,0 +1,82 @@
+using BugSplatUnity.Runtime.Reporter;
+using BugSplatUnity.Runtime.Settings;
+using NUnit.Framework;
+using System;
+using UnityEngine;
+
+namespace BugSplatUnity.RuntimeTests.Reporter
+{
+    public class ReportUploadGuardServiceTests
+    {
+        [Test]
+        public void ShouldPostException_WhenPostExceptionsInEditorFalse_WhenInEditor_ShouldReturnFalse()
+        {
+            var clientSettings = new WebGLClientSettingsRepository();
+            clientSettings.PostExceptionsInEditor = false;
+            clientSettings.ShouldPostException = (Exception ex) => true;
+            var rugs = new ReportUploadGuardService(clientSettings);
+            var exception = new Exception();
+
+            Assert.IsFalse(rugs.ShouldPostException(exception));
+        }
+
+        [Test]
+        public void ShouldPostException_WhenPostExceptionsInEditorTrue_WhenInEditor_ShouldReturnTrue()
+        {
+            var clientSettings = new WebGLClientSettingsRepository();
+            clientSettings.PostExceptionsInEditor = true;
+            clientSettings.ShouldPostException = (Exception ex) => true;
+            var rugs = new ReportUploadGuardService(clientSettings);
```

Comment:
Before merging I just wanted to mention how thrilled I am when a class name has a fun acronym

Reply:
😂🤣😂

Reply:
Thank you for pointing this out, I just burst out laughing 😆